src/main/java/ch/uzh/ifi/access/student/evaluation/evaluator/CodeEvaluator.java

```diff
@@ -47,14 +47,26 @@ public SubmissionEvaluation evaluate(StudentSubmission submission, Exercise exer
 		validate(submission, exercise);
 		CodeSubmission codeSub = (CodeSubmission) submission;
 
-		SubmissionEvaluation.Points scoredPoints = parseScoreFromLog(codeSub.getConsole().getEvalLog());
-		List<String> hints = parseHintsFromLog(codeSub.getConsole().getEvalLog());
+		String log = codeSub.getConsole().getEvalLog();
+
+		SubmissionEvaluation.Points scoredPoints = parseScoreFromLog(log);
+		List<String> hints = parseHintsFromLog(log);
+
+		if (exercise.getMaxScore() != scoredPoints.getMax()) {
```

Comment:
`Ponts.max` does not represent the max score of an exercise but rather the number of test cases in the suite if I'm not mistaken.

This check would then normally fail.

Reply:
I would recommend a name refactoring then to make this more clear. Why introduce fancy names when we could just name the thing... I would propose `Exercise.getMaxPoints` and `SubmissionEvaluation.TestResults` with two accessors `TestResults.getNumTestsPassing` and `TestResults.getNumTestsTotal`.

Reply:
I don't think this is the right place to discuss changes in the model. If you feel the naming is off, I suggest you open another ticket and we can take a look at this separately.